test/test_distributions.py

```diff
@@ -654,3 +656,185 @@ def von_mises(loc, concentration, value):
         actual = dist.VonMises(loc, concentration, d)(value=value)
     check_funsor(actual, inputs, reals())
     assert_close(actual, expected)
+
+
+def _check_sample(funsor_dist, sample_inputs, inputs, atol=1e-2, rtol=1e-2):
+    """utility that compares a Monte Carlo estimate of a distribution mean with the true mean"""
+    for tensor in list(funsor_dist.params.values())[:-1]:
+        tensor.data.requires_grad_()
+
+    sample_value = funsor_dist.sample(frozenset(['value']), sample_inputs)
+    expected_inputs = OrderedDict(
+        tuple(sample_inputs.items()) + tuple(inputs.items()) + (('value', funsor_dist.inputs['value']),)
+    )
+    check_funsor(sample_value, expected_inputs, reals())
+
+    if sample_inputs:
+
+        actual_mean = Integrate(
+            sample_value, Variable('value', funsor_dist.inputs['value']), frozenset(['value'])
+        ).reduce(ops.add, frozenset(sample_inputs))
+
+        inputs, tensors = align_tensors(*list(funsor_dist.params.values())[:-1])
+        expected_mean = Tensor(funsor_dist.dist_class(*tensors).mean, inputs)
+
+        check_funsor(actual_mean, expected_mean.inputs, expected_mean.output)
+        assert_close(actual_mean, expected_mean, atol=atol, rtol=rtol)
+
+        actual_variance = Integrate(
+            sample_value,
+            (Variable('value', funsor_dist.inputs['value']) - actual_mean) ** 2,
+            frozenset(['value'])
+        ).reduce(ops.add, frozenset(sample_inputs))
+        grad_targets = [v.data for v in list(funsor_dist.params.values())[:-1]]
+        actual_grads = torch.autograd.grad(actual_variance.reduce(ops.add).sum().data, grad_targets, allow_unused=True)
```

Comment:
Just curious, is there a reason you're computing gradients wrt variance rather than mean or other statistic?

Reply:
No particular reason, so I just switched to mean by default and added variance and entropy as options.